src/vs/workbench/api/common/extHostTerminalShellIntegration.ts

```diff
@@ -232,6 +240,16 @@ class InternalTerminalShellIntegration extends Disposable {
 		}
 	}
 
+	setEnv(keys: string[], values: string[]): void {
+		const env: { [key: string]: string | undefined } = {};
+		for (let i = 0; i < keys.length; i++) {
+			env[keys[i]] = values[i];
+		}
+		this._env = env;
+		// TODO: Make sure env changed. If env doesnt change, dont fire event.
```

Comment:
Let's do this, can we stop it at the capability time, rather than here or on the exthost?

Reply:
added them here: https://github.com/microsoft/vscode/pull/237415/commits/d5558133262f5d7cbce3610de096e21cf2decbcb 
at shellEnvDetectionCapability!

Reply:
👏